### PR TITLE
feat(uee): [IOPID-1325] add DataTableProfileEmailsRepository error cause

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

1. add `cause` field on error throws by `DataTableProfileEmailsRepository`'s `delete` and `insert`
2. remove `get` method

#### Motivation and Context

with this change we can achieve better error handling

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by silvicir — APPROVED** [review]

File: src/utils/unique_email_enforcement/storage.ts [hunk 1]
```diff
@@ -100,9 +89,15 @@ export class DataTableProfileEmailsRepository
...
       const entity = ProfileEmailToTableEntity.encode(p);
       await this.tableClient.deleteEntity(entity.partitionKey, entity.rowKey);
-    } catch {
+    } catch (e) {
       throw new Error(
-        `unable to delete the specified entity from ${this.tableClient.tableName} table`
+        `unable to delete the specified entity from ${this.tableClient.tableName} table`,
+        {
+          cause:
+            isRestError(e) && e.statusCode === 404
```
Comment (gquadrati) [inline src/utils/unique_email_enforcement/storage.ts:97]:
```suggestion
            isRestError(e) && 
            e.statusCode === 404 &&
            e.message.includes(`"ResourceNotFound"`)
```

File: src/utils/unique_email_enforcement/storage.ts [hunk 2]
```diff
@@ -100,9 +89,15 @@ export class DataTableProfileEmailsRepository
...
+              ? "RESOURCE_NOT_FOUND"
```
Comment (gquadrati) [inline src/utils/unique_email_enforcement/storage.ts:98]: Move all of them to exported const, so we can use it for checking the cause

**Review by gquadrati — APPROVED** [review]
LGTM except for the two comments

**Pull request merged**